jerry-debugger/src/lib/debugger-client.ts

```diff
@@ -23,27 +25,33 @@ export interface JerryDebuggerOptions {
 export const DEFAULT_DEBUGGER_HOST = 'localhost';
 export const DEFAULT_DEBUGGER_PORT = 5001;
 
-export class JerryDebugger {
+export class JerryDebuggerClient {
   readonly host: string;
   readonly port: number;
   readonly verbose: boolean;
+  private protocolHandler: JerryDebugProtocolHandler;
   private socket?: WebSocket;
+  private connectPromise?: Promise<void>;
 
   constructor(options: JerryDebuggerOptions) {
     this.host = options.host || DEFAULT_DEBUGGER_HOST;
     this.port = options.port || DEFAULT_DEBUGGER_PORT;
     this.verbose = options.verbose || false;
+    this.protocolHandler = new JerryDebugProtocolHandler(this);
```

Comment:
When you rework this, I'd probably inject the "transport" (websocket client) as well as the delegate when constructing the `JerryDebugProtocolHandler`. This makes it easier to test things as well as make it easier to use different kinds of transports as well.